pkg/apis/flowcontrol/types.go

```diff
@@ -133,14 +133,17 @@ type PriorityLevelConfigurationReference struct {
 }
 
 // PolicyRuleWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
-// making the request, the verb being requested, and the resource to be acted upon.
+// making the request, the verb being requested, and the resource or subresource or non-resource to be acted upon.
 type PolicyRuleWithSubjects struct {
 	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
 	// +optional
 	Subjects []Subject
-	// `rule` is the target verb, resource or the subresource the rule cares about. APIGroups, Resources, etc.
+	// `rule` is the target verb and most attributes of the resource or subresource or non-resource the rule cares about. This includes APIGroups, Resources, etc.  This excludes the namespace, if any, identifying the resource or subresource to act upon.  This exclusion is because the PolicyRule datatype is copied from rbac, which identifies the namespace in a different way.
 	// Required.
 	Rule PolicyRule
+	// `objectNamespace` identifies the namespace, if any, of the resource or subresource that the rule matches.  For a non-resource or non-namespaced resource this must be the empty string in order to match.
+	// +optional
+	ObjectNamespace string
```

Comment:
this should be a field under `Rule` of type `[]string` supporting wildcard. on the other hand, i think this new filed is sort of overlapping w/ `ByNamespace` distinguisher method. 

is this field is only meant for matching leader-election requests for now? matching the "coordination" resource should be sufficient for 1.16+ cluster

Reply:
Matching is distinct from flow distinguisher computation.

If you look at the KEP right now, it has two cases of discriminating on object namespace.  And more broadly, object namespace is a major part of the API data model, we should not omit the ability to discriminate on it.

Reply:
I thought we discussed this earlier and you said you prefer to keep PolicyRule as a copy of what is in rbac.  Putting the namespace up in PolicyRuleWithSubjects allows us to keep our PolicyRule a copy of rbac's.

Reply:
I have revised to make it a list of namespaces rather than a single one.
I am waiting for a reply on whether to move into PolicyRule.  I am OK with either way, just need an agreement.